funcx_endpoint/funcx_endpoint/executors/high_throughput/executor.py

```diff
@@ -776,6 +785,32 @@ def shutdown(self, hub=True, targets='all', block=False):
         logger.info("Finished HighThroughputExecutor shutdown attempt")
         return True
 
+    def _cancel(self, future):
+        logger.debug("Sending cancel of task_id:{future.task_id} to interchange")
+        response = self.command_client.run(TaskCancel(future.task_id))
+        logger.debug("Sent TaskCancel to interchange")
+        return response
+
+
+CANCELLED = 'CANCELLED'
+CANCELLED_AND_NOTIFIED = 'CANCELLED_AND_NOTIFIED'
+FINISHED = 'FINISHED'
+
+
+class FuncXFuture(Future):
```

Comment:
I'm unclear on long term architectural goals, but at least in the parsl style of modularisation this is a future related to htex, not to `FuncX`: switching to one of the other executors would mean using those other executors cancel (or lack of) functionality.